hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/constraints/WorkloadConstraints.java

```diff
@@ -0,0 +1,72 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload.constraints;
+
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.api.TopicName;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Collections.emptyMap;
+
+public class WorkloadConstraints {
+
+    private final Map<SubscriptionName, Constraints> subscriptionConstraints;
+    private final Map<TopicName, Constraints> topicConstraints;
+    private final int consumersPerSubscription;
+    private final int maxSubscriptionsPerConsumer;
+    private final int availableConsumers;
+
+    public WorkloadConstraints(Map<SubscriptionName, Constraints> subscriptionConstraints,
+                               Map<TopicName, Constraints> topicConstraints,
+                               int consumersPerSubscription,
+                               int maxSubscriptionsPerConsumer,
+                               int availableConsumers) {
+        this.subscriptionConstraints = subscriptionConstraints != null ? subscriptionConstraints : emptyMap();
+        this.topicConstraints = topicConstraints != null ? topicConstraints : emptyMap();
+        this.consumersPerSubscription = consumersPerSubscription;
+        this.maxSubscriptionsPerConsumer = maxSubscriptionsPerConsumer;
+        this.availableConsumers = availableConsumers;
+    }
+
+    public int getConsumersNumber(SubscriptionName subscriptionName) {
+        final int requiredConsumers = getSubscriptionConstraints(subscriptionName)
```

Comment:
the algorithm could be simplified so have no code duplication:

```
final int requiredConsumers = subscriptionConstraints.getOrDefault(subscriptionName, topicConstraints.get(subscriptionName.getTopicName()))
.map(Constraints::getConsumersNumber)
.orElse(0);
```

Reply:
Fixed